packages/lesswrong/server/scripts/convertImagesToCloudinary.ts

```diff
@@ -67,58 +86,124 @@ async function convertImagesInHTML(html: string, originDocumentId: string, urlFi
+    }
+  }
+  
+  return uniq(imageUrls);
+}
+
+function rewriteSrcset(srcset: string, urlMap: Record<string,string>): string {
+  const imageVariants = srcset.split(",").map(tok=>tok.trim());
+  const rewrittenImageVariants = imageVariants.map(variant => {
+    let tokens = variant.split(" ");
+    if (tokens[0] in urlMap)
+      tokens[0] = urlMap[tokens[0]];
+    return tokens.join(" ");
+  });
+  return rewrittenImageVariants.join(", ");
+}
+
 /**
- * Reupload all images in a post to Cloudinary, and create a new revision with the updated html
- * @param postId - The post to reupload images for
+ * Reupload all images in an object (post, tag, user, etc) to Cloudinary. For each editable
+ * field which had an image that needed reuploading, creates a new revision with the updated
+ * HTML.
+ *
+ * @param collectionName The collection that this object is in
+ * @param _id - The object to reupload images for
  * @param urlFilterFn - A function that takes a URL and returns true if it should be mirrored, by default all URLs are mirrored except those in getImageUrlWhitelist()
  * @returns The number of images that were mirrored
  */
-export async function convertImagesInPost(postId: string, urlFilterFn: (url: string) => boolean = () => true): Promise<number> {
+export async function convertImagesInObject(
+  collectionName: CollectionNameString,
+  _id: string,
+  fieldName = "contents",
+  urlFilterFn: (url: string)=>boolean = ()=>true
+): Promise<number> {
+  let totalUploaded = 0;
   try {
-    const post = await Posts.findOne({_id: postId});
-    if (!post) {
+    const collection = getCollection(collectionName);
+    const obj = await collection.findOne({_id});
+
+    if (!obj) {
       // eslint-disable-next-line no-console
-      console.error(`Cannot convert images in post ${postId}: invalid post ID`);
+      console.error(`Cannot convert images in ${collectionName}.${_id}: ID not found`);
       return 0;
     }
     
-    const latestRev = await getLatestRev(postId, "contents");
+    const latestRev = await getLatestRev(_id, fieldName);
     if (!latestRev) {
       // eslint-disable-next-line no-console
-      console.error(`Could not find a latest-revision for post ID: ${postId}`);
+      console.error(`Could not find a latest-revision for ${collectionName} ID: ${_id}`);
       return 0;
     }
     
-    const newVersion = await getNextVersion(postId, "patch", "contents", false);
+    const newVersion = await getNextVersion(_id, "patch", fieldName, false);
     const now = new Date();
     // NOTE: we use the post contents rather than the revision contents because we don't
     // create a revision for no-op edits (this is arguably a bug)
-    const oldHtml = post.contents.html;
-    const {count: uploadCount, html: newHtml} = await convertImagesInHTML(oldHtml, postId, urlFilterFn);
+    const oldHtml = obj[fieldName].html;
+    const {count: uploadCount, html: newHtml} = await convertImagesInHTML(oldHtml, _id, urlFilterFn);
```

Comment:
The `Converted ${uploadCount} images` message below isn't accurate anymore - would be nice to make that message clearer and/or more useful